taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+        else {
+            const newRenderedTranslations = [...renderedTranslations];
+            newRenderedTranslations.map((allTagsObj) => (allTagsObj['languageCode'] === key) ? 
+                (
+                    allTagsObj['tags'].push({'index': uuid.v4(), 'tag' : ''}),
+                    // Updated tags assigned for later use
+                    tagsToBeInserted = allTagsObj['tags'].map(el => (el.tag))
+                ) : allTagsObj
+            )
+            setRenderedTranslations(newRenderedTranslations) // Set state
+
+        }
+        // Set state of main NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject['tags_'+key] = tagsToBeInserted;
+            return newNodeObject
+        })
+    }
+
+    function handleDelete(key, index) {
+        let tagsToBeInserted = []
+        // State of "MainLang_renderedTranslations" is updated according to format used
+        if (key === nodeObject.main_language) {
+            const duplicateMainLang_renderedTranslations = mainLang_renderedTranslations.filter(obj => !(index === obj.index))
+            setMainLang_renderedTranslations(duplicateMainLang_renderedTranslations); // Set state
+
+            // Updated tags assigned for later use
+            tagsToBeInserted = duplicateMainLang_renderedTranslations.map(el => (el.tag))
+        }
+        // State of "renderedTranslations" is updated according to format used
+        else {
+            let newRenderedTranslations = []
+            renderedTranslations.forEach((allTagsObj) => {
+                if (allTagsObj['languageCode'] === key) {
+                    const unDeletedTags = allTagsObj['tags'].filter((tagObj) => !(tagObj.index === index));
+                    newRenderedTranslations.push({
+                        'languageCode' : key,
+                        'tags' : unDeletedTags
+                    })
+                    // Updated tags assigned for later use
+                    tagsToBeInserted = [...unDeletedTags].map(el => (el.tag));
+                }
+                else {
+                    newRenderedTranslations.push(allTagsObj)
+                }
+            })
+            setRenderedTranslations(newRenderedTranslations) // Set state
+        }
+        // Set state of main NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject['tags_'+key] = tagsToBeInserted;
+            return newNodeObject
+        })
     }
 
     return ( 
-        <Box sx={{ml: 4}} className="translations">
+        <Box sx={{ml: 4}}>
             {/* Title */}
-            <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+                <IconButton sx={{mt: 3.5, ml: 1, color: "#808080"}} onClick={handleOpen}>
```

Comment:
same comment with using custom colors. Please avoid that, if not possible to set it in the ThemeProvider, then at leas have those in a `const customColors = { }` somewhere so can be reused or easily tracked